server/container/guice/pom.xml

```diff
@@ -38,6 +38,7 @@
         <module>blob-memory-guice</module>
         <module>blob-objectstorage-guice</module>
         <module>cassandra-guice</module>
+        <module>cassandra-guice-common</module>
```

Comment:
I don't understand what is in cassandra-guice and what is in cassandra-guice-common. We should not use `common` when we can avoid it

Reply:
`cassandra-guice-common` should be renamed `cassandra-guice-modules`.

Would you porefer this?

Reply:
The problem is not about naming but about the content of the maven module.
Let's see:
```
./org/apache/james/modules/data/CassandraUsersRepositoryModule.java
./org/apache/james/modules/data/CassandraSieveRepositoryModule.java
./org/apache/james/modules/data/CassandraDomainListModule.java
./org/apache/james/modules/data/CassandraRecipientRewriteTableModule.java
./org/apache/james/modules/data/CassandraMailRepositoryModule.java
./org/apache/james/modules/data/CassandraJmapModule.java
./org/apache/james/modules/data/CassandraDLPConfigurationStoreModule.java
./org/apache/james/modules/mailbox/ElasticSearchClientModule.java
./org/apache/james/modules/mailbox/ResilientClusterProvider.java
./org/apache/james/modules/mailbox/CassandraQuotaModule.java
./org/apache/james/modules/mailbox/ElasticSearchQuotaSearcherModule.java
./org/apache/james/modules/mailbox/CassandraSessionModule.java
./org/apache/james/modules/mailbox/CassandraMailboxModule.java
./org/apache/james/modules/mailbox/TikaMailboxModule.java
./org/apache/james/modules/mailbox/ElasticSearchMailboxModule.java
./org/apache/james/modules/mailbox/ElasticSearchStartUpCheck.java
./org/apache/james/modules/mailbox/CassandraObjectStoreModule.java
./org/apache/james/modules/mailbox/CassandraSchemaVersionStartUpCheck.java
./org/apache/james/modules/mailbox/CassandraQuotaMailingModule.java
./org/apache/james/modules/mailbox/TikaConfigurationReader.java
./org/apache/james/modules/eventstore/CassandraEventStoreModule.java
./org/apache/james/modules/metrics/CassandraMetricsModule.java
./org/apache/james/CassandraModules.java
./org/apache/james/server/CassandraProbe.java
./org/apache/james/server/CassandraTruncateTableTask.java
```

We should:
* put data/ into a project of its own
* probably move DLP and Jmap (and maybe some more) out of data/
* put mailbox/ into a project of its own
* move a lot of stuff out of mailbox/ (ES, Tika, ResilientClusterProvider?)
* put eventstore/ into a project of its own
* split  CassandraModules in pieces
* maybe create a cassandra-guice-backend for CassandraMetricsModule, CassandraProbe
* move CassandraTruncateTableTask to test scope (it's a dangerous class, isn't it?)

Reply:
I agree to target such an orthogonal architecture.

However such changes are clearly too big regarding the effect desired (dropping the ActiveMQ dependency in distributed James) hence I refrained myself from doing it - I rather targeted a change unlocking the desired effect, which could be done by splitting modules from Guice products.

Reply:
I don't see why extracting the ActiveMQ code into its own guice module would not be enough then

Reply:
Because cassandra-guice defines the Cassandra Main which requires ActiveMQ. The distributed James depens on Cassandra Guice thus would inherit ActiveMQ anyway!

Reply:
(the rootcause is that we should not have twoproducts depending on each others...)

Reply:
Let's do nothing then. We'll fix that by killing Cassandra + ActiveMQ product